gateway/src/main/java/de/tum/in/www1/artemis/gateway/security/jwt/JWTFilter.java

```diff
@@ -1,5 +1,6 @@
 package de.tum.in.www1.artemis.gateway.security.jwt;
 
+import de.tum.in.www1.artemis.security.jwt.TokenProvider;
 import org.springframework.http.server.reactive.ServerHttpRequest;
 import org.springframework.security.core.Authentication;
```

Comment:
You wrote that previously reusing the JWTFilter didn't work but now it does. But it's still there. Could you please clarify this? I probably misunderstood you there.

Reply:
Sorry, I suppose I didn't explain clearly. 
Reusing the TokenProvider didn't work before for the gateway. JWTFilter cannot be reused in the Gateway because it has to implement a different interface due to the gateway using webflux. So the JWTFilter is there for the Gateway but is removed from User Management and Lecture.
The TokenProvider is removed from all 3